### PR TITLE
feat: Support ModelMetadata endpoint

### LINKED ISSUE #14: Support ModelMetadata endpoint
From the [KServe V2 Predict API](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md):

#### Model Metadata:

`GET v2/models/${MODEL_NAME}[/versions/${MODEL_VERSION}]`

Details of the REST API are [here](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#model-metadata) and corresponding gRPC API [here](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#model-metadata-1).

### PR DESCRIPTION
**Pull request by rafvasq**
#### Motivation
[KServe V2 API's ModelMetadata endpoint](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#model-metadata) is not yet supported in the `rest-proxy` grpc-gateway. 

#### Modifications
Extend grpc-gateway to include it in addition to [the `Infer` call already implemented.](https://github.com/kserve/rest-proxy/blob/079569e9499bfb7552f19192d29c3a31369f73b0/grpc_predict_v2.proto#L49-L55) 

#### Result
Resolves #14 and https://github.com/kserve/modelmesh-serving/issues/418

### CODE REVIEW

**Comment by Jooho** [comment]
simply tested this and I confirm that this work well

1. Without version
~~~
 curl -k https://example-onnx-mnist-modelmesh-serving.apps.jlee-test.oylv.p1.openshiftapps.com/v2/models/example-onnx-mnist
{"name":"example-onnx-mnist__isvc-b29c3d91f3", "versions":["1"], "platform":"OpenVINO", "inputs":[{"name":"Input3", "datatype":"FP32", "shape":["1", "1", "28", "28"]}], "outputs":[{"name":"Plus214_Output_0", "datatype":"FP32", "shape":["1", "10"]}]}  support-rest-endpoints  ~/temp/20230822_Test_repo/rest-proxy       
~~~

2. With version
~~~
curl -k https://example-onnx-mnist-modelmesh-serving.apps.jlee-test.oylv.p1.openshiftapps.com/v2/models/example-onnx-mnist/versions/1
{"name":"example-onnx-mnist__isvc-b29c3d91f3", "versions":["1"], "platform":"OpenVINO", "inputs":[{"name":"Input3", "datatype":"FP32", "shape":["1", "1", "28", "28"]}], "outputs":[{"name":"Plus214_Output_0", "datatype":"FP32", "shape":["1", "10"]}]}
~~~

File: go.mod [hunk 1]
```diff
@@ -5,27 +5,30 @@ go 1.18
...
 	go.uber.org/zap v1.24.0 // indirect
-	golang.org/x/net v0.3.1-0.20221206200815-1e63c2f08a10 // indirect
-	golang.org/x/sys v0.3.0 // indirect
-	golang.org/x/text v0.5.0 // indirect
-	google.golang.org/genproto v0.0.0-20221207170731-23e4bf6bdc37 // indirect
+	golang.org/x/net v0.14.0 // indirect
+	golang.org/x/sys v0.11.0 // indirect
+	golang.org/x/text v0.12.0 // indirect
+	google.golang.org/genproto v0.0.0-20230822172742-b8732ec3820d // indirect
+	google.golang.org/genproto/googleapis/api v0.0.0-20230822172742-b8732ec3820d // indirect
```
Comment (ckadner) [inline go.mod:30]: these additional dependencies should not be required

File: go.mod [hunk 2]
```diff
@@ -5,27 +5,30 @@ go 1.18
 require (
 	github.com/google/go-cmp v0.5.9
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.0
-	google.golang.org/grpc v1.51.0
-	google.golang.org/protobuf v1.28.1
+	google.golang.org/grpc v1.57.0
```
Comment (ckadner) [inline go.mod:8]: should we update those to the versions used by KServe?

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks @rafvasq -- need a few updates for the go.mod dependencies. I get different generated results, without these updates

**Comment by ckadner** [comment]
Thanks @rafvasq -- did you also regenerate the protos?

With `google.golang.org/grpc v1.51.0` I do get slightly different generated code

https://github.com/kserve/rest-proxy/pull/26/files#diff-2858165aafcdd2fe79ab516a84060fb8645cd00e44a92f623f3a57fe4b3b569cR446


image

**Review by ckadner — APPROVED** [review]
/lgtm

**Pull request merged**